### PR TITLE
feat: Modal.method support afterClose

### LINKED ISSUE #28007: Modal.method should support afterClose
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
If a Modal.method modal is closed with destroy() or destroyAll(), there is no way to run some code to deal with this case. Imagine a modal in a Promise resolved when the user click on the cancel or ok button, then it can lead to an unresolved Promise.

### What does the proposed API look like?
An afterClose option should be added to Modal.method.

### PR DESCRIPTION
**Pull request by liuchao233**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
close #28007 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Modal.method support afterClose |
| 🇨🇳 Chinese | Modal.method 支持 afterClose 属性 |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/modal/index.en-US.md](https://github.com/liuchao233/ant-design/blob/#28007/components/modal/index.en-US.md)
[View rendered components/modal/index.zh-CN.md](https://github.com/liuchao233/ant-design/blob/#28007/components/modal/index.zh-CN.md)

### CODE REVIEW

**Review by afc163 — APPROVED** [review]

File: components/modal/confirm.tsx [hunk 1]
```diff
@@ -81,7 +81,12 @@ export default function confirm(config: ModalFuncProps) {
     currentConfig = {
       ...currentConfig,
       visible: false,
-      afterClose: destroy.bind(this, ...args),
+      afterClose: () => {
+        if (typeof config.afterClose === 'function') {
+          config.afterClose();
```
Comment (afc163) [inline components/modal/confirm.tsx:86]: 测试没覆盖到这一行。

**Review by afc163 — CHANGES_REQUESTED** [review]

File: components/modal/index.zh-CN.md [hunk 2]
```diff
@@ -67,7 +67,11 @@ cover: https://gw.alipayobjects.com/zos/alicdn/3StSdUlSH/Modal.svg
 
 | 参数 | 说明 | 类型 | 默认值 | 版本 |
 | --- | --- | --- | --- | --- |
+<<<<<<< HEAD
+| afterClose | Modal 完全关闭后的回调 | function | - |  |
+=======
 | afterClose | Modal 完全关闭后的回调 | function | - | 4.9.0 |
+>>>>>>> a8c9b571edd85b6609a8771ed150f456556bd620
```
Comment (afc163) [inline components/modal/index.zh-CN.md:74]: conflict

Comment (liuchao233) [reply components/modal/index.zh-CN.md]: sorry, my fault

File: components/modal/index.en-US.md [hunk 3]
```diff
@@ -64,6 +64,11 @@ The items listed above are all functions, expecting a settings object as paramet
 
 | Property | Description | Type | Default | Version |
 | --- | --- | --- | --- | --- |
+<<<<<<< HEAD
+| afterClose | Specify a function that will be called when modal is closed completely | function | - |  |
+=======
+| afterClose | Specify a function that will be called when modal is closed completely | function | - | 4.9.0 |
+>>>>>>> a8c9b571edd85b6609a8771ed150f456556bd620
```
Comment (afc163) [inline components/modal/index.en-US.md:71]: Please check changes carefully.

**Review by afc163 — APPROVED** [review]

**Pull request merged**